packages/legend-query/src/components/QueryBuilderPropertyExpressionEditor.tsx

```diff
@@ -87,21 +96,65 @@ const DerivedPropertyParameterValueEditor = observer(
     );
     const resetParameterValue = (): void => {
       const genericType = guaranteeNonNullable(variable.genericType);
-      const primitiveInstanceValue = new PrimitiveInstanceValue(
-        genericType,
-        variable.multiplicity,
-      );
-      if (genericType.value.rawType.name !== PRIMITIVE_TYPE.LATESTDATE) {
-        primitiveInstanceValue.values = [
-          generateDefaultValueForPrimitiveType(
-            genericType.value.rawType.name as PRIMITIVE_TYPE,
+      const _type = genericType.value.rawType;
```

Comment:
can we do something about this code duplication? The shared logic here is where we need to create a new value specification for a parameter yeah?